File: python-pr/52/python/code-quality-readability-maintainability/report_generator_18.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def generate_report(data):
    report = "REPORT\n" + "="*50 + "\n"
    total_items = len(data)
    report += f"Total Items: {total_items}\n"
    if total_items > 0:
        numeric_fields = [k for k, v in data[0].items() if isinstance(v, (int, float))]
        for field in numeric_fields:
            values = [item[field] for item in data if field in item]
            if values:
                avg = sum(values) / len(values)
                report += f"Average {field}: {avg:.2f}\n"
    return report
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/report_generator_18.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function generateReport(data) {
    let report = "REPORT\n" + "=".repeat(50) + "\n";
    let totalItems = data.length;
    report += `Total Items: ${totalItems}\n`;
    if (totalItems > 0) {
        let numericFields = Object.keys(data[0]).filter(k => typeof data[0][k] === 'number');
        for (let field of numericFields) {
            let values = data.map(item => item[field]).filter(v => typeof v === 'number');
            if (values.length > 0) {
                let avg = values.reduce((sum, v) => sum + v, 0) / values.length;
                report += `Average ${field}: ${avg.toFixed(2)}\n`;
            }
        }
    }
    return report;
}
// {/fact}